paddle/fluid/imperative/layer.h

```diff
@@ -59,7 +59,8 @@ class VarBase {
   static std::vector<std::string> AliveVarNames();
   explicit VarBase(bool has_grad, const std::string& name)
       : name_(name),
-        grad_var_(has_grad ? new VarBase(false, GradVarName()) : nullptr) {
+        grad_var_(has_grad ? new VarBase(false, GradVarName()) : nullptr),
+        is_grad_from_grad_maker_(false) {
```

Comment:
A bit hard code. I guess you want to distinguish whether the var is a grad var or not. Try to use some thing like `grad_to_var` (the type can be `std::unorderd_map<VarBase* /*grad_var*/, VarBase* /*fwd_var*/>`) to distinguish whether the var is a grad var.

Reply:
std::unorderd_map<VarBase* /*grad_var*/, VarBase* /*fwd_var*/> is a slower way to slove this thing